luhn_algorithm.rb

```diff
@@ -0,0 +1,79 @@
+account_number = ARGV.first
+
+class Card_number
+  def initialize(entered_number)
+    @array_elements = []
+    @entered_number = entered_number
+    @counter_array = 0
+    @array_size = @entered_number.length
+    @counter_calculations = 0
+    @sum1 = 0
+    @sum2 = 0
+    @sum3 = 0
+  end
+
+  def valid?
+    digits_array
+    calculations
+    calculations_result
+    check_result
+  end
+
+  private
+
+  def digits_array
+    @array_elements = @entered_number.split(//)
+    @array_elements = @array_elements.reverse
+  end

```

Comment:
Това може да се опрости дори още повече:

```suggestion
  def digits_array
    @array_elements = @entered_number.split(//).reverse
  end

```

Reply:
В този метод можеш да преобразуваш всичките елементи с `to_i`, вместо да правиш това навсякъде в кода.